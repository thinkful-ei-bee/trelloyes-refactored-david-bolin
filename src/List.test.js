import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';

describe('List Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header="x" cards={[1, 2]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List header="First List" cards={['a', 'b']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});
