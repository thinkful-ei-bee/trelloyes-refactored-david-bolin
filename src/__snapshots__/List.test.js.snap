// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
